### PR TITLE
PYMT-1663: Common Health checks.

### PR DESCRIPTION
**Pull request by rashmitsingh**
- DatabaseHealthCheck.
- QueueHealthCheck
- RedisHealthCheck
- DynamoDbHealthCheck
- ClientConnectivityHealthCheck

### CODE REVIEW

File: src/Health/Checks/ClientConnectivityHealthCheck.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    private $name;
+
+    /**
+     * ClientConnectivityHealthCheck constructor.
+     *
+     * @param \GuzzleHttp\ClientInterface $client
+     * @param string $endpoint Endpoint to hit. This needs to be a GET method.
+     * @param string $serviceName The name of service you are connecting to. Used when displaying service health.
+     */
+    public function __construct(ClientInterface $client, string $endpoint, string $serviceName)
```
Comment (rashmitsingh) [inline src/Health/Checks/ClientConnectivityHealthCheck.php]: I am thinking of this specific health check as a generic check which can be used to test for multiple external clients. Thus passing in all these arguments to be able to configure it.

File: src/Health/Checks/DatabaseHealthCheck.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+     * {@inheritdoc}
+     */
+    public function check(): HealthState
+    {
+        $state = HealthInterface::STATE_HEALTHY;
+
+        try {
+            $this->entityManager->getConnection()->executeQuery('SELECT true');
+        } /** @noinspection BadExceptionsProcessingInspection */ catch (Exception $exception) {
+            // If exception is thrown database is not readable, ignore this and set the state to degraded
```
Comment (merk) [inline src/Health/Checks/DatabaseHealthCheck.php]: I think we should only catch Doctrine exceptions here, if some other exception gets thrown it isnt talking about doctrine

File: tests/Health/Checks/ClientConnectivityHealthCheckTest.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     * Test check returns degraded status when connection fails.
+     *
+     * @return void
+     */
+    public function testClientConnectionFails(): void
+    {
+        $client = new FailingClientStub();
+        $instance = new ClientConnectivityHealthCheck(
+            $client,
+            '\health',
```
Comment (merk) [inline tests/Health/Checks/ClientConnectivityHealthCheckTest.php]: `\health` ? Shouldnt it be `/health` ?

Comment (rashmitsingh) [reply tests/Health/Checks/ClientConnectivityHealthCheckTest.php]: it for sure should be

**Pull request merged**